### DISCUSSION TITLE
TC420App

### DISCUSSION THREAD

**Discussion by stf888** [schwabix-1311/aquaPI#14] [Ideas]
Hey Thomas, I am an author of the TC420App ( https://tc420app.ubrothers.eu/ ) which supports various versions of TC420 (USB) and TC421 (USB/WiFi). Let me know if you need some help protocol-wise etc. Your project is very nice, if you are planning to add more LED dimming options using MOSFET/PWM, I am could add support for this in my app if you want :) I am also playing with Raspberry PI, having few ideas for my fish room - like increasing light brightness when I enter the room, monitoring air pump using pressure sensor and turning backup pump on in case of failure etc.. Anyway, good job!

**Comment by schwabix-1311** [comment]
Hi "stf888",

I'm Markus, the author of aquaPi. The aquaPi software includes a SW driver to use the TC420 directly via USB, as an alternative to PWM (ARM internal or from a I²C chip) plus MOSFET power drivers. In this mode there's no other software like your TC420App required.

All dimming effects are generated by aquaPi software on the Raspberry CPU and sent to the TC420, using what the device calls "Fast Play" mode. Please see my fork of Adam Wallner's TC420 tools; I had to add some capabilities and fix a few glitches.

Although aquaPi is not quite ready for the public, it should already be able to configure a backup pump if you have a pressure sensor providing either a switch signal from a threshold, or an analog pressure signal as input for a MinimumNode.

Likewise, your brightness boost could be achieved with a motion sensor going into a GPIO, which is then read by a SwitchNode; OR-ed with the output of the regular light control node(s). A MaximumNode exists for exactly such use cases.

A feature on my ToDo is the ability to read light profiles from TC420 applications (your's is not the only one I've seen) and let aquaPi run these profiles on whatever PWM output. In this scenario TC420App would be an nice addition to aquaPi in that it allows graphical edits.  Is your file format documented?

Markus
